### ISSUE TITLE
Revisit offline check in net

### ISSUE THREAD

**Issue by jfroelich** [jfroelich/rss-reader#762]
When offline, an awaited call to `fetch` against an invalid url throws a `TypeError` with the message `Failed to fetch`. This TypeError indicates a network error, e.g. connectivity, e.g. no network.

When online, an awaited call to `fetch` against an invalid url returns a response with status 404.

When calling `fetch`, it is possible to call it with bad parameters. When calling fetch with bad parameters, it rejects with a type error. Unlike the other situation, here a type error means typeerror, indicating bad parameters to the function.

I assume that if I know I will be using valid parameters, then I can treat all type errors indication of network errors. this means no need to check for offline state explicitly.

In other words:
* remove the offline check
* trap the error thrown by the native fetch call
* throw instead a networkerror or fetcherror or some kind of custom error
* update all callers to no longer rely on offlineerror and instead infer fetcherror/networkerror means the same thing
